src/file_chooser_dialog.rs

```diff
@@ -5,20 +5,31 @@
 use ffi;
 use glib::translate::*;
 use glib::object::{Downcast, IsA};
+use libc::c_char;
 use std::ptr;
 use FileChooserAction;
 use FileChooserDialog;
+use ResponseType;
 use Widget;
 use Window;
 
 impl FileChooserDialog {
-    pub fn new<T: IsA<Window>>(title: Option<&str>, parent: Option<&T>, action: FileChooserAction)
-            -> FileChooserDialog {
+    pub fn new<T: IsA<Window>>(title: Option<&str>, parent: Option<&T>, action: FileChooserAction, buttons: &[(&str, ResponseType)]) -> FileChooserDialog {
         assert_initialized_main_thread!();
         unsafe {
-            Widget::from_glib_none(
-                ffi::gtk_file_chooser_dialog_new(title.to_glib_none().0, parent.to_glib_none().0,
-                    action.to_glib(), ptr::null_mut()))
+            Widget::from_glib_none(match buttons.len() {
+                0 => ffi::gtk_file_chooser_dialog_new(title.to_glib_none().0, parent.to_glib_none().0, action.to_glib(), ptr::null::<c_char>()),
+                1 => ffi::gtk_file_chooser_dialog_new(title.to_glib_none().0, parent.to_glib_none().0, action.to_glib(), buttons[0].0.to_glib_none().0, buttons[0].1.to_glib(), ptr::null::<c_char>()),
+                2 => {
+                    let second_button_text: *const c_char = buttons[1].0.to_glib_none().0;
```

Comment:
This won't work.
You need save `Stash` to variable and use `.0` in function call.

Reply:
Thanks, I'll fix it.

Reply:
Fixed in the latest commit